spec/compile_spec.rb

```diff
@@ -1,10 +1,94 @@
 require_relative "spec_helper"
 require "fileutils"
 
+require "fontist"
+
 RSpec.describe Metanorma::Compile do
+  def clean_outputs
+    %w(xml presentation.xml html alt.html doc relaton.xml err rxl pdf).each do |ext|
+      FileUtils.rm_f Dir["spec/assets/*.#{ext}"]
+    end
+
+    FileUtils.rm_rf "spec/assets/test"
+    FileUtils.rm_rf "spec/assets/extract"
+  end
+
+  before(:each) do clean_outputs end
+  after(:all) do clean_outputs end
+
+  it "fontist_install called" do
+    compile = Metanorma::Compile.new
+
+    allow(compile).to receive(:fontist_install) {}
+    expect(compile).to receive(:fontist_install).once
+
+    compile.compile("spec/assets/test.adoc", type: "iso", :"agree-to-terms" => true)
+  end
+
+  it "fontist_install called with explicit no_install_fonts=false" do
+    compile = Metanorma::Compile.new
+
+    allow(compile).to receive(:fontist_install) {}
+    expect(compile).to receive(:fontist_install).once
+
+    compile.compile("spec/assets/test.adoc", type: "iso", :"agree-to-terms" => true, :"no-install-fonts" => false)
```

Comment:
@CAMOBAP can we change the signature of the `compile` method to:
```ruby
compile.compile("spec/assets/test.adoc", type: "iso", cli_args: {:"agree-to-terms" => true, :"no-install-fonts" => false})
```
?

I believe in the future we will have per-flavor CLI options too.